sympy/tensor/array/array_comprehension.py

```diff
@@ -328,4 +327,54 @@ def tomatrix(self):
         if self._rank != 2:
             raise ValueError('Dimensions must be of size of 2')
 
-        return Matrix(self._expand_array())
+        return Matrix(self._expand_array().tomatrix())
+
+
+def isLambda(v):
+    LAMBDA = lambda: 0
+    return isinstance(v, type(LAMBDA)) and v.__name__ == LAMBDA.__name__
+
+class ArrayComprehensionMap(ArrayComprehension):
+    '''
+    A subclass of ArrayComprehension dedicated to map external function lambda.
+
+    Notes
+    =====
+
+    Only the lambda function is considered.
+    At most one argument in lambda function is accepted in order to avoid ambiguity.
+
+    Examples
+    ========
+
+    >>> from sympy.tensor.array import ArrayComprehensionMap
+    >>> from sympy import symbols
+    >>> i, j, k = symbols('i j k')
+    >>> a = ArrayComprehensionMap(lambda: 1, (i, 1, 4))
+    >>> a.doit()
+    [1, 1, 1, 1]
+    >>> b = ArrayComprehensionMap(lambda a: a+1, (j, 1, 4))
+    >>> b.doit()
+    [2, 3, 4, 5]
+
+    '''
+    def __new__(cls, function, *symbols, **assumptions):
```

Comment:
maybe `limits` instead of `symbols`?

Reply:
In fact we have discussed before, during the creation of `ArrayComprehension`. It was `limits` in the first place. But in the end we agreed that this class should follow the example of `ExprWithLimits` whose constructor use `function`, `symbols` and `assumptions` as arguments.